### PR TITLE
Blocks: Add missing `wp-block-woocommerce-{name}` to Add to Cart with Options, Product Image, Product Rating, Product Rating Stars, Product Rating Counter and Product Image

### PR DESCRIPTION
**Pull request by thealexandrelara**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Part of #49739 .

This PR adds the missing `wp-block-woocommerce-{name}` to the following blocks:

- Product Image
- Product Rating
- Product Rating Stars
- Product Rating Counter
- Add to Cart with Options

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Visit the shop page (/shop) and click on a product.
2. After being redirected to the Single Product page, make sure the Product Image, Add to Cart with Options, Product Rating, Product Rating Counter and Product Ratings Stars blocks are being rendered correctly.
    2.1. To test the Product Rating Stars and Product Rating Counter, follow these instructions:
    2.2. Go to a template/post/page;
    2.3. Add the Single Product block. 
    2.4. Switch to the "Code Editor" mode (in MacOS: Shift + Option + CMD + M);
    2.5. Add the following within the Single Product block:

         ```
          <!-- wp:woocommerce/product-rating-counter {"isDescendentOfSingleProductBlock":true} /-->

          <!-- wp:woocommerce/product-rating-stars {"isDescendentOfSingleProductBlock":true} /-->
          ```

5. Save and check the changes on the frontend.
3. Inspect each block and make sure you can see the `wp-block-woocommerce-{block-name}` attached to the class property of the HTML tag. (`{block_name}` is the name of the block, for instance for Product Rating, it is `product-rating`). 

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Add missing `wp-block-woocommerce-{name}` class to Add to Cart with Options, Product Image, Product Rating, Product Rating Stars, Product Rating Counter and Product Image blocks.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php [hunk 1]
```diff
@@ -98,12 +98,16 @@ protected function render( $attributes, $content, $block ) {
...
 
+		$wrapper_attributes = get_block_wrapper_attributes(
+			array(
+				'class' => 'wc-block-add-to-cart-form ' . esc_attr( $classes_and_styles['classes'] ) . ' ' . esc_attr( $classname ) . ' ' . esc_attr( $product_classname ),
+				'style' => esc_attr( $classes_and_styles['styles'] ),
+			)
+		);
+
 		$form = sprintf(
-			'<div class="wp-block-add-to-cart-form wc-block-add-to-cart-form %1$s %2$s %3$s" style="%4$s">%5$s</div>',
```
Comment (louwie17) [inline plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php]: `wp-block-add-to-cart-form` was there a specific reason for removing `wp-block-add-to-cart-form` and only keeping the `wc-block-add-to-cart-form`?

File: plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php [hunk 2]
```diff
@@ -98,12 +98,16 @@ protected function render( $attributes, $content, $block ) {
 		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
 		$product_classname  = $is_descendent_of_single_product_block ? 'product' : '';
 
+		$wrapper_attributes = get_block_wrapper_attributes(
+			array(
+				'class' => 'wc-block-add-to-cart-form ' . esc_attr( $classes_and_styles['classes'] ) . ' ' . esc_attr( $classname ) . ' ' . esc_attr( $product_classname ),
```
Comment (louwie17) [inline plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php]: I think we can remove this one `esc_attr( $classname )`, as it already gets included as part of `esc_attr( $classes_and_styles['classes'] )`. At-least when I tested it and and added a custom class name in the site editor.

Comment (louwie17) [reply plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php]: Would it also make sense to use `array_filter` here as some of these classes can actually be empty causing a bunch of spaces: `wc-block-add-to-cart-form    wp-block-woocommerce-add-to-cart-form` 
While it doesn't break anything, it looks a bit odd when inspecting the html.
We use `array_filter` for several of the other block types within `get_block_wrapper_attributes`

**Review by louwie17 — CHANGES_REQUESTED** [review]
Thanks for this @thealexandrelara, this overall tested well although I wasn't sure how to get the Product Rating Counter and the Product Rating Star blocks to display, they don't show up for me, only the **Product Rating** block seems to show up for me. The changes for those blocks look fine, but would like to also test it.
Screenshot 2024-09-23 at 3 13 30 PM

I also noticed the removal of one of the classes in the add to cart form, removing it may be fine, but just wanted to call it out. Also suggesting to use a `filter` for with a `implode` to concatenate the classes together to avoid unnecessary whitespaces.

File: plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php (hunk 2 above)
Comment (thealexandrelara) [reply plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php]: Nice catch! I removed the `esc_attr( $classname )` and used the  `array_filter` method there (38d31719bac91bd2b0a967b551feaea2552beffa).

File: plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php (hunk 1 above)
Comment (thealexandrelara) [reply plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php]: Now that you mentioned, I just added it back (38d31719bac91bd2b0a967b551feaea2552beffa). I initially thought it was redundant but I just realized it does not contain the `woocommerce` namespace (`wp-block-add-to-cart-form !== wp-block-woocommerce-add-to-cart-form`).

**Comment by thealexandrelara** [comment]
Thank you for reviewing the PR, @louwie17!

>  I wasn't sure how to get the Product Rating Counter and the Product Rating Star blocks to display, they don't show up for me, only the Product Rating block seems to show up for me.

Yeah, I believe those blocks cannot be directly added to the template. One way to test them is to add them manually. 

1. Go to a template/post/page;
2. Add the Single Product block. 
3. Switch to the "Code Editor" mode (in MacOS: Shift + Option + CMD + M);
4. Add the following within the Single Product block:

```
<!-- wp:woocommerce/product-rating-counter {"isDescendentOfSingleProductBlock":true} /-->

<!-- wp:woocommerce/product-rating-stars {"isDescendentOfSingleProductBlock":true} /-->
```

5. Save and check the changes on the frontend.

>  I also noticed the removal of one of the classes in the add to cart form, removing it may be fine, but just wanted to call it out. Also suggesting to use a filter for with a implode to concatenate the classes together to avoid unnecessary whitespace.

Nice catch! I replied to your comment on the code and also changed the code to include the `implode` + `array_filter` methods.

Could you review the PR again?

**Review by louwie17 — APPROVED** [review]
Thanks for the updates @thealexandrelara and for adding the extra test instructions around the product rating blocks.
It would be worth updating the PR description as well to include those instructions for when this gets tested by QA.

Approving this though, code looks good and it tested well 💯

**Comment by gigitux** [comment]
I think that this PR introduced this regression: https://github.com/woocommerce/woocommerce/issues/51883

The point is that `get_block_wrapper_attributes` already generates all the necessary CSS classes and styles starting from attributes. This means that we should not pass to `get_block_wrapper_attributes` classed and styles generated with `StyleAttributesUtils`. Tomorrow I will work on a fix.

**Comment by thealexandrelara** [comment]
> I think that this PR introduced this regression: #51883
> 
> The point is that `get_block_wrapper_attributes` already generates all the necessary CSS classes and styles starting from attributes. This means that we should not pass to `get_block_wrapper_attributes` classed and styles generated with `StyleAttributesUtils`. Tomorrow I will work on a fix.

Ohh, TIL! Good to know that this removes the need for the StyleAttributesUtils.

**Pull request merged**